ReadingList/ReadingListSQLStorage.swift

```diff
@@ -94,6 +95,13 @@ class ReadingListSQLStorage: ReadingListStorage {
                 return Result(failure: ReadingListStorageError("Can't get first item from results"))
             }
         } else {
+            if let reason = statement.reason {
+                // TODO: this is ugly string matching, but checking for error SQLITE_CONSTRAINT does not say which constraint was violated, and even with checking for SQLITE_CONSTRAINT_UNIQUE extended result code, it is not possible to get which column's uniqueness was violated; so no idea at an "exact" way currently
```

Comment:
I will see if I can dig up the last SQLite error. It was available as a private member of `Statement`, which I can't access, I will look if I can access it publicly in `Database`.